cli-plugins/manager/hooks.go

```diff
@@ -0,0 +1,128 @@
+package manager
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/docker/cli/cli-plugins/hooks"
+	"github.com/docker/cli/cli/command"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// HookPluginData is the type representing the information
+// that plugins declaring support for hooks get passed when
+// being invoked following a CLI command execution.
+type HookPluginData struct {
+	RootCmd string
+	Flags   map[string]string
+}
+
+// RunPluginHooks calls the hook subcommand for all present
+// CLI plugins that declare support for hooks in their metadata
+// and parses/prints their responses.
+func RunPluginHooks(dockerCli command.Cli, rootCmd, subCommand *cobra.Command, plugin string, args []string) error {
+	subCmdName := subCommand.Name()
+	if plugin != "" {
+		// TODO: handle plugin alias (build = buildx)
```

Comment:
is this `TODO` for a follow up PR? Maybe link an issue if the intent is to do it in the _future_?

Reply:
Good callout! I put that there to remember to take care of it and then forgot 😅 glad you caught it. I'll create a ticket and clean it up.